Broker_Portals/portals.lua

```diff
@@ -222,92 +222,95 @@ local function SetupSpells()
-    tinsert(portals, { 1777023, 'TRUE' }) -- Yojamba Isle
-
-    if expac == "TBC" then
-      tinsert(portals, { 777016, 'TRUE' }) -- Shattrath
-      tinsert(portals, { 777017, 'TRUE' }) -- Area 52
-      tinsert(portals, { 777018, 'TRUE' }) -- Altar of Sha'tar
-      tinsert(portals, { 777019, 'TRUE' }) -- Sanctum of the Stars
-      tinsert(portals, { 102182, 'TRUE' }) -- Cenarion Refuge
-      tinsert(portals, { 102186, 'TRUE' }) -- Ogri'la
-      tinsert(portals, { 102196, 'TRUE' }) -- Stormspire
-      tinsert(portals, { 777008, 'TRUE' }) -- Sanctum of the Stars
-      tinsert(portals, { 777008, 'TRUE' }) -- Altar of Sha'tar
-      tinsert(portals, { 102180, 'TRUE' }) -- Cenarion Refuge
-
-        if UnitFactionGroup('player') == "Horde" then
-          tinsert(portals, { 777014, 'TRUE' }) -- Silvermoon City
-          tinsert(portals, { 102197, 'TRUE' }) -- Thrallmar
-          tinsert(portals, { 102189, 'TRUE' }) -- Shadowmoon Village
-          tinsert(portals, { 102184, 'TRUE' }) -- Garadar
-          tinsert(portals, { 102190, 'TRUE' }) -- Stonebreaker Hold
-          tinsert(portals, { 102201, 'TRUE' }) -- Zabra'jin
-        elseif UnitFactionGroup('player') == "Alliance" then
-          tinsert(portals, { 777015, 'TRUE' }) -- The Exodar
-          tinsert(portals, { 102185, 'TRUE' }) -- Honor Hold
-          tinsert(portals, { 102193, 'TRUE' }) -- Telaar
-          tinsert(portals, { 102178, 'TRUE' }) -- Allerian Stronghold
-          tinsert(portals, { 102187, 'TRUE' }) -- Orebor Harborage
-          tinsert(portals, { 102200, 'TRUE' }) -- Wildhammer Stronghold
-        end
+  end
 
+  -- Ascension: Scrolls of Defense
+  tinsert(portals, { 83126, 'TRUE' }) -- Ashenvale
+  tinsert(portals, { 83128, 'TRUE' }) -- Hillsbrad Foothills
+  -- Ascension: Runes of Retreat
+  local runes = {
+    { 979807 }, -- Flaming
+    { 80133 }, -- Frostforged
+    { 979806 }, -- Arcane
+    { 979808 }, -- Freezing
+    { 979809 }, -- Dark Rune
+    { 979810 } -- Holy Rune
+  }
+  local runeRandom = {}
+  for _, v in ipairs(runes) do
+    if IsSpellKnown(v[1]) then
+      tinsert(runeRandom, v[1])
     end
-
-    -- Ascension: Scrolls of Defense
-    tinsert(portals, { 83126, 'TRUE' }) -- Ashenvale
-    tinsert(portals, { 83128, 'TRUE' }) -- Hillsbrad Foothills
-    -- Ascension Rune of Retreat
-      local runes = {
-        { 979807 }, --Flaming Rune of Retreat
-        { 80133 }, --Frostforged Rune of Retreat
-        { 979806 }, --Arcane Rune of Retreat
-        { 979808 }, --Freezing Rune of Retreat
-        { 979809 }, --Dark Rune of Retreat
-        { 979810 }  --Holy Rune of Retreat
-      }
-      local runeRandom = {}
-        for _,v in ipairs(runes) do
-          if IsSpellKnown(v[1]) then
-            tinsert(runeRandom, v[1])
-          end
-        end
-        tinsert(portals, { runeRandom[math.random(1, #runeRandom)], 'TRUE' })
-    spells = nil
+  end
+  if runeRandom == {} then
```

Comment:
This will always be false.

use `if next(runeRandom)` or `if #runeRandom > 0`

Reply:
you're right 😅 
replaced it & pushed